master/buildbot/test/unit/test_process_builder.py

```diff
@@ -215,55 +213,128 @@ def test_getCollapseRequestsFn_builder_False(self):
         startable = yield self.bldr.canStartBuild('worker', 100)
         self.assertEqual(startable, True)
 
         startable = yield self.bldr.canStartBuild('worker', 101)
         self.assertEqual(startable, True)
 
-        # set a configurable one
-        record = []
+    @defer.inlineCallbacks
+    def test_canStartBuild_config_canStartBuild_returns_value(self):
+        yield self.makeBuilder()
 
         def canStartBuild(bldr, worker, breq):
-            record.append((bldr, worker, breq))
             return (worker, breq) == ('worker', 100)
+        canStartBuild = mock.Mock(side_effect=canStartBuild)
+
         self.bldr.config.canStartBuild = canStartBuild
 
         startable = yield self.bldr.canStartBuild('worker', 100)
         self.assertEqual(startable, True)
-        self.assertEqual(record, [(self.bldr, 'worker', 100)])
+        canStartBuild.assert_called_with(self.bldr, 'worker', 100)
+        canStartBuild.reset_mock()
 
         startable = yield self.bldr.canStartBuild('worker', 101)
         self.assertEqual(startable, False)
-        self.assertEqual(
-            record, [(self.bldr, 'worker', 100), (self.bldr, 'worker', 101)])
+        canStartBuild.assert_called_with(self.bldr, 'worker', 101)
+        canStartBuild.reset_mock()
 
-        # set a configurable one to return Deferred
-        record = []
+    @defer.inlineCallbacks
+    def test_canStartBuild_config_canStartBuild_returns_deferred(self):
+        yield self.makeBuilder()
 
-        def canStartBuild_deferred(bldr, worker, breq):
-            record.append((bldr, worker, breq))
+        def canStartBuild(bldr, worker, breq):
             return defer.succeed((worker, breq) == ('worker', 100))
-        self.bldr.config.canStartBuild = canStartBuild_deferred
+        canStartBuild = mock.Mock(side_effect=canStartBuild)
+
+        self.bldr.config.canStartBuild = canStartBuild
 
         startable = yield self.bldr.canStartBuild('worker', 100)
         self.assertEqual(startable, True)
-        self.assertEqual(record, [(self.bldr, 'worker', 100)])
+        canStartBuild.assert_called_with(self.bldr, 'worker', 100)
+        canStartBuild.reset_mock()
 
         startable = yield self.bldr.canStartBuild('worker', 101)
         self.assertEqual(startable, False)
-        self.assertEqual(
-            record, [(self.bldr, 'worker', 100), (self.bldr, 'worker', 101)])
+        canStartBuild.assert_called_with(self.bldr, 'worker', 101)
+        canStartBuild.reset_mock()
+
+    @defer.inlineCallbacks
+    def test_canStartBuild_cant_acquire_locks_but_no_locks(self):
+        yield self.makeBuilder()
+
+        self.bldr.botmaster.getLockFromLockAccess = \
+            mock.Mock(return_value='dummy')
+
+        with mock.patch(
+                'buildbot.process.build.Build._canAcquireLocks',
+                mock.Mock(return_value=False)):
+            startable = yield self.bldr.canStartBuild('worker', 100)
+            self.assertEqual(True, startable)
```

Comment:
why it is reversed?

I hope you're not one of those who prefer to write

```c++
if (0 == value){
/* "Yoda conditionals" ("If zero the value is, execute this code you must!") */
}
```

Reply:
Agreed. I've just used C# testing frameworks a lot and there the comparison order is `expected`, `actual`.

Reply:
Fixed now.